File: delivery-service/delivery/db/src/main/java/org/example/delivery/db/storeuser/StoreUserRepository.java
package org.example.delivery.db.storeuser;

import org.example.delivery.db.storeuser.enums.StoreUserStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface StoreUserRepository extends JpaRepository<StoreUserEntity, Long> {

    /**
     * SELECT * FROM store_user WHERE email = ? AND status = ? ORDER BY id DESC LIMIT 1
     */
    Optional<StoreUserEntity> findFirstByEmailAndStatusOrderByIdDesc(String email, StoreUserStatus status);
}

File: delivery/db/src/main/kotlin/org/example/delivery/db/storeuser/StoreUserRepository.kt
package org.example.delivery.db.storeuser

import org.example.delivery.db.storeuser.enums.StoreUserStatus
import org.springframework.data.jpa.repository.JpaRepository

interface StoreUserRepository : JpaRepository<StoreUserEntity, Long> {
    /**
     * SELECT * FROM store_user WHERE email = ? AND status = ? ORDER BY id DESC LIMIT 1
     */
    fun findFirstByEmailAndStatusOrderByIdDesc(email: String?, status: StoreUserStatus?): StoreUserEntity?
}
